packages/calypso-e2e/src/lib/components/sidebar-component.ts

```diff
@@ -63,19 +74,26 @@ export class SidebarComponent extends BaseContainer {
 	 * @param {{[key: string]: string}} param0 Named object parameter.
 	 * @param {string} param0.item Plaintext representation of the top level heading.
 	 * @param {string} param0.subitem Plaintext representation of the child level heading.
-	 * @throws {Error} If neither item or subitem were specified in the parameter.
 	 * @returns {Promise<void>} No return value.
 	 */
 	async gotoMenu( { item, subitem }: { item?: string; subitem?: string } ): Promise< void > {
 		let selector;
+		const viewport_name = getViewportName();
+
+		// If mobile, sidebar is hidden by default and focus is on the content.
+		// The sidebar must be first brought into view.
+		if ( viewport_name === 'mobile' ) {
+			await this._openSidebar();
```

Comment:
I think this check would make more sense inside the `openSidebar`, i.e. 

```js
async _openSidebarIfNecessary() {
	if ( viewportName !== 'mobile' ) {
		return;
	}
}
```

Reply:
I disagree; I think the method `openSidebar` should do only that, open the sidebar. 

I personally prefer to have methods and functions be named definitively instead of uncertainly. So in this case, `openSidebar` is very clear in what it does - it opens the sidebar. `openSidebarIfNecessary` leaves open room for interpretation on what 'necessary' means.

Reply:
`*IfNecessary` is a convention that is also used in Playwright, i.e. `ElementHandle.scrollToIfNeeded()` (a bit different wording there though). I understand the uncertainty issue though, so maybe at this point going with `openMobileSidebar` would be the best of both worlds?

Reply:
> *IfNecessary is a convention that is also used in Playwright, i.e. ElementHandle.scrollToIfNeeded()

To my knowledge that's the only time such naming convention is used in Playwright - and I fully understand that sometimes it's just not possible to be 100% consistent 😃 

> going with openMobileSidebar would be the best of both worlds?

Sure. I am not 100% on board with this name either because say a decision is made in the future that `tablet` viewport will also have hidden sidebar by default. The name then is no longer reflective of reality when the more catch-all name `openSidebar` would require no changes. 

With that said, that's a bridge we can cross when that time comes, so for now I can rename this method to reference the mobile viewport.

Reply:
>  I am not 100% on board with this name either because say a decision is made in the future that tablet viewport will also have hidden sidebar by default. The name then is no longer reflective of reality when the more catch-all name openSidebar would require no changes.

Good point! This way it definitely scales better and makes more sense from the user's perspective. Feel free to ignore my comment then 👍

Reply:
To be clear, I will stick with the renaming to `_openMobileSidebar` for now since the hypothetical situation I mentioned is unlikely to come to pass :)